### PR TITLE
Improve markdown logic

### PR DESCRIPTION
**Pull request by KKKZOZ**
## Before:
Before
## After:
After

### CODE REVIEW

File: src/components/Lotion.vue [hunk 1]
```diff
@@ -95,6 +95,17 @@ function setBlockType (blockIdx: number, type: BlockType) {
...
+  // the block should first turn into a text block
+  if([BlockType.H1, BlockType.H2, BlockType.H3,BlockType.Quote]
+      .includes(props.page.blocks[blockIdx].type)){
+    setBlockType(blockIdx, BlockType.Text)
+    setTimeout(()=>{
+      blockElements.value[blockIdx].moveToStart()
+    })
+    return
+  }
+
```
Comment (greentfrapp) [inline src/components/Lotion.vue:108]: This looks good but can we move it to above line 96? Otherwise it wouldn't apply to the first block.

Comment (KKKZOZ) [reply src/components/Lotion.vue:106]: That's right!

Comment (greentfrapp) [reply src/components/Lotion.vue:106]: LGTM!

**Review by greentfrapp — APPROVED** [review]
LGTM thanks @KKKZOZ! 🧴

**Pull request merged**